x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/create_event_signal.ts

```diff
@@ -92,12 +92,31 @@ export const createEventSignal = async ({
       indexFields: threatIndexFields,
     };
 
-    const signalsQueryMap = await getSignalsQueryMapFromThreatIndex({
-      threatSearchParams,
-      eventsCount: currentEventList.length,
-      signalValueMap: getSignalValueMap({ eventList: currentEventList, threatMatchedFields }),
-      termsQueryAllowed: true,
-    });
+    let signalsQueryMap;
+    try {
+      signalsQueryMap = await getSignalsQueryMapFromThreatIndex({
+        threatSearchParams,
+        eventsCount: currentEventList.length,
+        signalValueMap: getSignalValueMap({
+          eventList: currentEventList,
+          threatMatchedFields,
+        }),
+        termsQueryAllowed: true,
+      });
+    } catch (exc) {
+      // we receive an error if the event list count < threat list count
+      // which puts us into the create_event_signal which differs from create threat signal
+      // in that we call getSignalsQueryMapFromThreatIndex which can *throw* an error
+      // rather than *return* one.
+      if (
+        exc.message.includes('Query contains too many nested clauses; maxClauseCount is set to')
```

Comment:
I agree it would be a good idea to add the `failed to create query: maxClauseCount is set to` as a clause in that catch block. I haven't run into this specifically which is surprising. Do you mind sharing the steps you took to test this? Based on the rule details I'm assuming you used the kbn-alert-load lib. What were the settings you used there?

Reply:
I just followed steps you provided in the ticket's description earlier for `kbn-alert-load` tool

```
./kbn-alert-load.js run --type='im' -e http://elastic:changeme@localhost:9200 -k http://elastic:changeme@localhost:5601/kbn im-test    

```

Reply:
Hmm.. any possible changes to docker? Are you using an M2 Macbook or Intel?